docs/docs/adding-an-rss-feed.md

```diff
@@ -0,0 +1,157 @@
+- Your content isn't in Markdown so the plugin doesn't know about it
+- Your Markdown files have dates in the filenames, for which the slug URLs cause 404s
+
+The good news is you can accommodate these scenarios and more in `gatsby-config.js` and `gatsby-node.js`.
+
+To customize the default feed schema (a.k.a. structure) output by the plugin to work with your website's content, you can start with the following code:
+
+```json:title=gatsby.config.js
+{
+      resolve: `gatsby-plugin-feed`,
+      options: {
+        query: `
+          {
+            site {
+              siteMetadata {
+                title
+                description
+                siteUrl
+                site_url: siteUrl
+              }
+            }
+          }
+        `,
+        feeds: [
+          {
+            /* highlight-start */
+            serialize: ({ query: { site, allMarkdownRemark } }) => {
+              return allMarkdownRemark.edges.map(edge => {
+                /* highlight-end */
+                return Object.assign({}, edge.node.frontmatter, {
+                  description: edge.node.excerpt,
+                  date: edge.node.frontmatter.date,
+                  url: site.siteMetadata.siteUrl + edge.node.fields.slug,
+                  guid: site.siteMetadata.siteUrl + edge.node.fields.slug,
+                  custom_elements: [{ "content:encoded": edge.node.html }],
+                })
+              })
+            },
+            query: `
+              {
+                // highlight-next-line
+                allMarkdownRemark(
+                  sort: { order: DESC, fields: [frontmatter___date] },
+                ) {
+                  edges {
+                    node {
+                      excerpt
+                      html
+                      fields { slug }
+                      frontmatter {
+                        title
+                        date
+                      }
+                    }
+                  }
+                }
+              }
+            `,
+            output: "/rss.xml",
+            title: "Your Site's RSS Feed",
+          },
+        ],
+      },
+    },
+```
+
+This snippet contains a custom `gatsby-plugin-feed` setup in `gatsby-config.js` to query metadata for your site, like its `title` and `siteUrl`. It also includes a `feeds` array with at least one object containing a GraphQL query and `serialize` method, which allows you to output a custom RSS feed structure. In this example, the RSS content comes from Markdown files sourced from your site, and queried with the key `allMarkdownRemark` and its associated filters and fields.
+
+The `output` field in your feed object allows you to customize the filename for your RSS feed, and `title` for the name of your site's RSS feed.
+
+> To see your feed in action, run `gatsby build && gatsby serve` and you can then inspect the content and URLs in your RSS file at `http://localhost:9000/rss.xml`.
+
+## Additional customization for content slugs
```

Comment:
This seems a little bit tangential? (and the regex is a little gnarly 😈)

Reply:
I don't think my setup is that unusual, and it seems like something worth mentioning so people don't have to figure it out on their own? I'm open to regex wizardry suggestions 😄

Reply:
Generally - I think if we document how to create a slug (which we have!), then the rest will sort of follow.

So if someone knew they wanted to remove dates from their slugs--since they already know _how_ to create a slug, that part would be pretty understandable?

Not sure--I can see how there's value in documenting extra cases, but I also don't know how common this is. Guessing you're doing this for your site?